tasks/elasticsearch-config.yml

```diff
@@ -4,54 +4,101 @@
 #Create conf directory
 - name: Create Configuration Directory
   become: yes
-  file: path={{ es_conf_dir }} state=directory owner=root group={{ es_group }} mode=2750
+  file:
+    path: "{{ es_conf_dir }}"
+    state: directory
+    owner: root
+    group: "{{ es_group }}"
+    mode: "2750"
 
 #Create pid directory
 - name: Create PID Directory
   become: yes
-  file: path={{ es_pid_dir }} state=directory owner={{ es_user }} group={{ es_group }} mode=0755
+  file:
+    path: "{{ es_pid_dir }}"
+    state: directory
+    owner: "{{ es_user }}"
+    group: "{{ es_group }}"
+    mode: "755"
 
 #Create required directories
 - name: Create Others Directories
   become: yes
-  file: path={{ item }} state=directory owner={{ es_user }} group={{ es_group }} mode=2750
+  file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ es_user }}"
+    group: "{{ es_group }}"
+    mode: "2750"
   with_items:
     - "{{ es_log_dir }}"
     - "{{ es_data_dirs }}"
 
 #Copy the config template
 - name: Copy Configuration File
   become: yes
-  template: src=elasticsearch.yml.j2 dest={{ es_conf_dir }}/elasticsearch.yml owner=root group={{ es_group }} mode=0660 force=yes
+  template:
+    src: elasticsearch.yml.j2
+    dest: "{{ es_conf_dir }}/elasticsearch.yml"
+    owner: root
+    group: "{{ es_group }}"
+    mode: "660"
+    force: yes
   register: system_change
   notify: restart elasticsearch
 
 #Copy the default file
 - name: Copy Default File
   become: yes
-  template: src=elasticsearch.j2 dest={{ default_file }} owner=root group={{ es_group }} mode=0660 force=yes
+  template:
+    src: "{% if es_config_default != '' %}{{ es_config_default }}{% else %}elasticsearch.j2{% endif %}"
+    dest: "{{ default_file }}"
+    owner: root
+    group: "{{ es_group }}"
+    mode: "660"
+    force: yes
   notify: restart elasticsearch
 
 #Copy the systemd specific file if systemd is installed
 - when: use_system_d and m_lock_enabled
   become: yes
   block:
     - name: Make sure destination dir exists
-      file: path={{ sysd_config_file | dirname }} state=directory mode=0755
+      file:
+        path: "{{ sysd_config_file | dirname }}"
+        state: directory
+        mode: "755"
 
     - name: Copy specific ElasticSearch Systemd config file
-      ini_file: path={{ sysd_config_file }} section=Service option=LimitMEMLOCK value=infinity mode=0644
+      ini_file:
+        path: "{{ sysd_config_file }}"
+        section: Service
+        option: LimitMEMLOCK
+        value: infinity
+        mode: "644"
       notify:
        - reload systemd configuration
        - restart elasticsearch
 
-#Copy the logging.yml
-- name: Copy log4j2.properties File
+- name: Copy jvm.options File
   become: yes
-  template: src={{ es_config_log4j2 }} dest={{ es_conf_dir }}/log4j2.properties owner=root group={{ es_group }} mode=0660 force=yes
+  template:
+    src: "{% if es_config_jvm != '' %}{{ es_config_jvm }}{% else %}jvm.options.j2{% endif %}"
```

Comment:
Does this do something different than just setting a default value? So having: `es_config_jvm: "jvm.options.j2"` in `defaults/main.yml`.

Reply:
no differences, fixed in https://github.com/elastic/ansible-elasticsearch/pull/637/commits/ea4d6cf43a8114be9e061ddc5e10afa1c1607a72